server/container/guice/guice-common/src/test/java/org/apache/james/MailsShouldBeWellReceived.java

```diff
@@ -89,6 +94,45 @@ default void mailsShouldBeWellReceived(GuiceJamesServer server) throws Exception
 
     }
 
+    @Test
+    default void mailsShouldBeWellReceivedByBothRecipient(GuiceJamesServer server) throws Exception {
```

Comment:
1. can you add tests about one hundred mails to multiple recipients? 
2. Two maybe is not great, many more recipients might trigger something different.  Can you try?

Reply:
You are right. It fails if we send on mail to more than 2 recipients...
And with multiple causes : cassandra concurrent exception when saving the message to all the recipients,
interrupted exception in StoreMessageManager.createAndDispatchMessage
, null pointer during spamassassin

Reply:
I added a test with 10 recipients. It's slow but it works. The errors i reported previously where due to test timeout.